spot_driver/spot_driver/robot_command_util.py

```diff
@@ -0,0 +1,267 @@
+        return [command]
+
+    # This is the increment to find the position of the next batch.
+    stride = batch_size - overlapping
+    if stride < 1:
+        return [command]
+
+    index = 0
+    commands: List[robot_command_pb2.RobotCommand] = []
+
+    is_last_batch = False
+    while not is_last_batch:
+        is_last_batch = True
+
+        new_command = robot_command_pb2.RobotCommand()
+        new_command.CopyFrom(command)
+
+        if new_command.HasField("synchronized_command"):
+            if new_command.synchronized_command.HasField("mobility_command"):
+                mobility_request = new_command.synchronized_command.mobility_command
+                trajectory = mobility_request.se2_trajectory_request.trajectory
+                is_last_batch = slice_trajectory(trajectory, index, batch_size) and is_last_batch
+            if new_command.synchronized_command.HasField("arm_command"):
+                arm_request = new_command.synchronized_command.arm_command
+                if arm_request.HasField("arm_cartesian_command"):
+                    trajectory = arm_request.arm_cartesian_command.pose_trajectory_in_task
+                    is_last_batch = slice_trajectory(trajectory, index, batch_size) and is_last_batch
+                elif arm_request.HasField("arm_joint_move_command"):
+                    trajectory = arm_request.arm_joint_move_command.trajectory
+                    is_last_batch = slice_trajectory(trajectory, index, batch_size) and is_last_batch
+                elif arm_request.HasField("arm_impedance_command"):
+                    trajectory = arm_request.arm_impedance_command.task_tform_desired_tool
+                    is_last_batch = slice_trajectory(trajectory, index, batch_size) and is_last_batch
+            if new_command.synchronized_command.HasField("gripper_command"):
+                gripper_request = new_command.synchronized_command.gripper_command
+                trajectory = gripper_request.claw_gripper_command.trajectory
+                is_last_batch = slice_trajectory(trajectory, index, batch_size) and is_last_batch
+
+        commands.append(new_command)
+        index += stride
+
+    return commands
+
+
+def min_time_since_reference(command: robot_command_pb2.RobotCommand) -> float:
+    """
+    This method inspects all trajectories in the command and returns the minimum
+    time_since_reference.
+    It returns float("inf") if no trajectory is available.
+
+    Args:
+        command: The command to inspect.
+
+    Returns:
+        The minimum time_since_reference of all trajectories in the command.
+    """
+    min_time_since_reference = float("inf")
+    if command.HasField("synchronized_command"):
+        if command.synchronized_command.HasField("mobility_command"):
+            mobility_request = command.synchronized_command.mobility_command
+            if trajectory := mobility_request.se2_trajectory_request.trajectory:
```

Comment:
Are we sure we want to be using this `:=` assignment operator inside the conditional? I thought this would simply always return a `True` because it's just outputting the result of the trajectory assignment, as there's no conditional attached to it.

Reply:
In the if condition you highlighted, first the `mobility_request.se2_trajectory_request.trajectory` is evaluated. If it is true (i.e. is not None), its content is assigned to the local variable `trajectory` and the code inside the if condition is executed.
If the condition was always true, even in the scenario where the trajectory is empty, trying to access `points[0]` will throw an exception.